Bug: Random Panic on Activation

I got a random crash when running `hermit init`, seemingly in the progress bar drawing code. I'll try to look into this later, time permitting. It seemed to successfully initialize hermit, so maybe this crash was "benign?"
```
njackson@nick-comp:~/crum/squeal$ hermit init 
info: Creating new Hermit environment in /home/njackson/crum/squeal
info:   -> /home/njackson/crum/squeal/bin/activate-hermit
info:   -> /home/njackson/crum/squeal/bin/activate-hermit.fish
info:   -> /home/njackson/crum/squeal/bin/hermit
info:   -> /home/njackson/crum/squeal/bin/README.hermit.md
info:   -> /home/njackson/crum/squeal/bin/hermit.hcl
info: 

Hermit environment initialised in /home/njackson/crum/squeal

To activate the environment run:

  . /home/njackson/crum/squeal/bin/activate-hermit

Then run the following to list available commands:

  hermit --help

To deactivate the environment run:

  deactivate-hermit

For more information please refer to https://github.com/cashapp/hermit

Hermit environment /home/njackson/crum/squeal activated
panic: strings: negative Repeat count [recovered]
        panic: strings: negative Repeat count

goroutine 1 [running]:
github.com/cashapp/hermit/app.Main.func3()
	/home/runner/work/hermit/hermit/app/main.go:147 +0x56
panic({0xa4adc0?, 0xc26ea0?})
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/runtime/panic.go:792 +0x132
strings.Repeat({0xb381c3?, 0x6?}, 0xc0002dd7f8?)
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/strings/strings.go:624 +0x585
github.com/cashapp/hermit/ui.(*UI).writeProgress(0xc000266700, 0xd3)
	/home/runner/work/hermit/hermit/ui/ui.go:279 +0x28b
github.com/cashapp/hermit/ui.(*UI).redrawProgress(0xc000266700)
	/home/runner/work/hermit/hermit/ui/ui.go:229 +0xc5
github.com/cashapp/hermit/ui.(*Task).Add(0xc000266100, 0x0?)
	/home/runner/work/hermit/hermit/ui/task.go:74 +0xa8
github.com/cashapp/hermit/ui.(*progressWriter).Write(0xc0004563e0?, {0xc0003d2000?, 0x8000, 0x8000?})
	/home/runner/work/hermit/hermit/ui/task.go:94 +0x25
io.(*teeReader).Read(0xc000456400, {0xc0003d2000, 0x47525f?, 0x8000})
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/io/io.go:630 +0x6e
io.copyBuffer({0xc29aa0, 0xc00031e168}, {0xc29da0, 0xc000456400}, {0x0, 0x0, 0x0})
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/io/io.go:429 +0x190
io.Copy(...)
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/io/io.go:388
os.genericReadFrom(0xc00031e140?, {0xc29da0, 0xc000456400})
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/os/file.go:185 +0x4f
os.(*File).ReadFrom(0xc00031e140, {0xc29da0, 0xc000456400})
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/os/file.go:161 +0x9c
io.copyBuffer({0xc29940, 0xc00031e140}, {0xc29da0, 0xc000456400}, {0x0, 0x0, 0x0})
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/io/io.go:415 +0x151
io.Copy(...)
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/io/io.go:388
github.com/cashapp/hermit/cache.downloadHTTP(0xc000462e70?, 0xc00017cc60, {0xc0003d0bd4, 0x40}, {0xc000038f40, 0x31}, {0xc0000427e0, 0x82})
	/home/runner/work/hermit/hermit/cache/http.go:117 +0x858
github.com/cashapp/hermit/cache.(*httpSource).Download(0xc0004cd278, 0xc000302c00, 0x31?, {0xc0003d0bd4, 0x40})
	/home/runner/work/hermit/hermit/cache/http.go:45 +0x305
github.com/cashapp/hermit/cache.(*Cache).Download(0xc000462ed0, 0xc000302c00, {0xc0003d0bd4, 0x40}, {0xc000038f40?, 0xc00030d201?}, {0x115aea0?, 0x50bbd3?, 0xffffffffffffff9c?})
	/home/runner/work/hermit/hermit/cache/cache.go:128 +0x69b
github.com/cashapp/hermit/state.(*State).extract(0xc0004af380, 0xc000302c00, 0xc0000cb908)
	/home/runner/work/hermit/hermit/state/state.go:399 +0x20e
github.com/cashapp/hermit/state.(*State).CacheAndUnpack(0xc0004af380, 0xc000302c00, 0xc0000cb908)
	/home/runner/work/hermit/hermit/state/state.go:315 +0x152
github.com/cashapp/hermit.(*Env).install(0xc000402240, 0xc000266700, 0xc0000cb908)
	/home/runner/work/hermit/hermit/env.go:728 +0x306
github.com/cashapp/hermit.(*Env).Install(0xc000402240, 0xc000266700, 0xc0000cb908)
	/home/runner/work/hermit/hermit/env.go:643 +0x693
github.com/cashapp/hermit/app.(*installCmd).Run(0xc0003101d8, 0xc000266700, 0xc000402240, 0xc0004af380)
	/home/runner/work/hermit/hermit/app/install_cmd.go:95 +0x94b
reflect.Value.call({0xa84f20?, 0xc0003101d8?, 0xb36ec0?}, {0xb3870a, 0x4}, {0xc000325380, 0x3, 0x4?})
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/reflect/value.go:584 +0xca6
reflect.Value.Call({0xa84f20?, 0xc0003101d8?, 0xc0001110b0?}, {0xc000325380?, 0xb36ec0?, 0xd5631326ddc40106?})
	/home/runner/.cache/hermit/pkg/go-1.24.5/src/reflect/value.go:368 +0xb9
github.com/alecthomas/kong.callAnyFunction({0xa84f20?, 0xc0003101d8?, 0xa85a20?}, 0xc0004b8f00)
	/home/runner/go/pkg/mod/github.com/alecthomas/kong@v1.12.1/callbacks.go:225 +0x7b8
github.com/alecthomas/kong.callFunction({0xa84f20?, 0xc0003101d8?, 0xc000111338?}, 0xc0004b8f00)
	/home/runner/go/pkg/mod/github.com/alecthomas/kong@v1.12.1/callbacks.go:182 +0x17b
github.com/alecthomas/kong.(*Context).RunNode(0xc000302300, 0xc00032d1d0, {0xc000111bb8, 0xa, 0xa})
	/home/runner/go/pkg/mod/github.com/alecthomas/kong@v1.12.1/context.go:846 +0x5a7
github.com/alecthomas/kong.(*Context).Run(0xc000302300, {0xc000111bb8?, 0x0?, 0x5?})
	/home/runner/go/pkg/mod/github.com/alecthomas/kong@v1.12.1/context.go:875 +0x13c
github.com/cashapp/hermit/app.Main({{0xc000278380, 0x10}, 0x3, {0xc00024a600, 0x3a}, {0xc000257a10, 0x3, 0x3}, {0xc00023ac00, 0x1a, ...}, ...})
	/home/runner/work/hermit/hermit/app/main.go:316 +0x20f9
main.main()
	/home/runner/work/hermit/hermit/cmd/hermit/main.go:28 +0x218

```

--- a/cache/http.go
+++ b/cache/http.go
@@ -107,7 +107,7 @@ func downloadHTTP(b *ui.Task, response *http.Response, checksum string, uri stri
 		return "", "", "", errors.WithStack(err)
 	}
 	resumed := info.Size()
-	task.Size(int(max(response.ContentLength + resumed, 0)))
+	task.Size(int(max(response.ContentLength+resumed, 1)))
 	task.Add(int(resumed))
 	defer task.Done()
 

--- a/ui/ui.go
+++ b/ui/ui.go
@@ -272,7 +272,7 @@ func (w *UI) writeProgress(width int) {
 	columns := int(float64(width-15) * float64(barsn) * percent)
 	nofm := fmt.Sprintf("%d/%d", complete, len(liveOperations))
 	percentstr := fmt.Sprintf("%.1f%%", percent*100)
-	spaces := max(width - columns/barsn - 15, 0)
+	spaces := max(width-columns/barsn-15, 0)
 	fmt.Fprintf(w.stdout, "%s%s%s %-7s%6s\n", strings.Repeat(theme.fill, max(columns/barsn, 0)), theme.bars[columns%barsn], strings.Repeat(theme.blank, spaces), nofm, percentstr)
 	// Write operations bar.
 	for _, op := range liveOperations {

